### PR TITLE
added mouse wheel/scroll to `nimgame2/input.nim`

### PR DESCRIPTION
**Pull request by CodeDoes**

### CODE REVIEW

**Comment by Vladar4** [comment]
Good idea!

A couple of notes though:

1. It looks like `event.wheel.x` and `event.wheel.y` return only 1, or -1 values. Might be handier to add event's values to the `mWheel` variable and add its reset to `(0, 0)` into `proc initMouse()`.

2. Maybe add automatic `-1` multiplication based on the `event.wheel.direction` value? See: https://wiki.libsdl.org/SDL_MouseWheelEvent?highlight=%28%5CbCategoryStruct%5Cb%29%7C%28CategoryEvents%29#Remarks

Let me know if you're up to implement these additions, otherwise I'll merge your commit and continue myself.

P.S.: Could you please keep PascalCase constants' names for the consistency?

Thanks!

**Comment by CodeDoes** [comment]
Thanks. I implemented it for my own use and thought I'd contribute. 

I'll implement the additions.

**Comment by CodeDoes** [comment]
The wiki is ambiguos examples would have been appreciated. I think the mousewheelflipped bool is just to tell you if the mousewheel is flipped. I could be wrong though

File: nimgame2/input.nim [hunk 1]
```diff
@@ -193,6 +194,7 @@ proc initMouse*() =
   ##
   mPressed = 0
   mReleased = 0
+  mWheel = (0.0,0.0)
```
Comment (CodeDoes) [inline nimgame2/input.nim:198]: I already have the reset code

**Comment by Vladar4** [comment]
> The wiki is ambiguos examples would have been appreciated. I think the mousewheelflipped bool is just to tell you if the mousewheel is flipped. I could be wrong though

"If direction is SDL_MOUSEWHEEL_FLIPPED the values in x and y will be opposite. Multiply by -1 to change them back."

As we are not saving the `direction` value, we could use the `mWheel` as "normal" by multiplying values from events with "flipped" `direction`.

> I already have the reset code

Now you need to add new event's values to the `mWheel`. Otherwise it'll always be `-1`, `0`, or `1`, regardless of the number of MouseWheel events since the last update.

**Comment by CodeDoes** [comment]
I already had the reset code for mouse wheel in `initMouse`
I added some changes

**Comment by Vladar4** [comment]
Looks good as a starting point. I'll perform some testing and tuning later myself.

Thanks for the help!

**Comment by CodeDoes** [comment]
Oh. Stack the values!

**Comment by Vladar4** [comment]
Yeah, I've noticed. Will fix it during the testing stage too.

**Pull request merged**